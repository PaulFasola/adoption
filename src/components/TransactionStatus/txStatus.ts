export enum TxStatus {
	COMPLETED = 'completed',
	PENDING = 'pending',
	UNKNOWN = 'unknown',
	FAILED = 'failed'
}
