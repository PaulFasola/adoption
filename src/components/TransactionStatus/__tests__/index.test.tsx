import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import { advanceTo, clear } from 'jest-date-mock';
import { render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { configure, shallow } from 'enzyme';
import { TxStatus } from '../txStatus';
import { TransactionStatus } from '..';

expect.extend(toHaveNoViolations);
configure({ adapter: new Adapter() });

describe('TransactionStatus component', () => {
  it('should render', () => {
    shallow(<TransactionStatus amount="0.0001" symbol="BTC" status={TxStatus.PENDING} />)
  });

  it('should render the tx date, when provided', () => {
    advanceTo(new Date(2009, 1, 3, 0, 0, 0));
    const wrapper = shallow(<TransactionStatus amount="0.0001" symbol="BTC" status={TxStatus.PENDING} date={{ value: new Date() }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    clear();
  });

  it('should render a date in the specified locale', () => {
    advanceTo(new Date(2009, 1, 3, 0, 0, 0));
    const wrapper = shallow(<TransactionStatus amount="0.0001" symbol="BTC" status={TxStatus.PENDING} date={{ value: new Date(), locale: 'fr-FR' }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    clear();
  });

  const ComplexTx = (<TransactionStatus
    animated
    autoShowDetails
    amount="0.1"
    symbol="BTC"
    status={TxStatus.COMPLETED}
    txURL="https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1"
    txFees="0.005"
    sender={{
      hash: "1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa",
      url: "https://www.blockchain.com/btc/address/1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa"
    }}
    receiver={{
      hash: "1CounterpartyXXXXXXXXXXXXXXXUWLpVr",
      url: "https://www.blockchain.com/btc/address/1CounterpartyXXXXXXXXXXXXXXXUWLpVr"
    }}
    date={{
      value: new Date(2009, 0, 9, 10, 42)
    }}
  />);

  it('should render a complex form', () => {
    const wrapper = shallow(ComplexTx);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should not have any accessibility issues', async () => {
    const { container } = render(ComplexTx);
    expect(await axe(container)).toHaveNoViolations();
  });

});