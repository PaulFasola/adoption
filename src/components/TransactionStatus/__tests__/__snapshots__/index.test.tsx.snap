// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionStatus component should render a complex form 1`] = `
<styled.div
  hasDate={true}
  showDetails={false}
>
  <styled.div
    animate="completed"
    backgroundColor="grey"
  />
  <styled.div
    showSide={true}
  >
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": "0.1",
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="<h1>{amount}</h1><p>{symbol}</p>"
      />
    </styled.div>
    <styled.div
      title="Status"
    >
      UNKNOWN
    </styled.div>
    <styled.div>
      <styled.div>
        <Icon
          style={
            Object {
              "width": 21,
            }
          }
          targetBlank={true}
          type="outbound-link"
          url="https://www.blockchain.com/btc/tx/369d241af595fc253479abe394e2f21fda05820a0416942f63266dd793035cf1"
        />
      </styled.div>
      <styled.div
        clickable={true}
        onClick={[Function]}
      >
        <Icon
          style={
            Object {
              "width": 15,
            }
          }
          type="arrow-down"
        />
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <div>
      <styled.div>
        <a
          href="https://www.blockchain.com/btc/address/1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa"
          rel="noreferrer"
          target="_blank"
        >
          1A1zP1e...7DivfNa
        </a>
        <styled.div />
        <a
          href="https://www.blockchain.com/btc/address/1CounterpartyXXXXXXXXXXXXXXXUWLpVr"
          rel="noreferrer"
          target="_blank"
        >
          1Counte...XUWLpVr
        </a>
      </styled.div>
      <styled.div>
        0.005
         
        BTC
         (fees)
      </styled.div>
    </div>
    <styled.div />
  </styled.div>
  <styled.div>
    <span>
      1/9/2009
    </span>
  </styled.div>
</styled.div>
`;

exports[`TransactionStatus component should render a date in the specified locale 1`] = `
<styled.div
  hasDate={true}
  showDetails={false}
>
  <styled.div
    animate={null}
    backgroundColor="grey"
  />
  <styled.div
    showSide={false}
  >
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": "0.0001",
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="<h1>{amount}</h1><p>{symbol}</p>"
      />
    </styled.div>
    <styled.div
      title="Status"
    >
      UNKNOWN
    </styled.div>
  </styled.div>
  <styled.div>
    <span>
      03/02/2009
    </span>
  </styled.div>
</styled.div>
`;

exports[`TransactionStatus component should render the tx date, when provided 1`] = `
<styled.div
  hasDate={true}
  showDetails={false}
>
  <styled.div
    animate={null}
    backgroundColor="grey"
  />
  <styled.div
    showSide={false}
  >
    <styled.div>
      <AdaptiveSpan
        mapping={
          Array [
            Object {
              "style": "bold",
              "tag": "{amount}",
              "value": "0.0001",
            },
            Object {
              "tag": "{symbol}",
              "value": "BTC",
            },
          ]
        }
        text="<h1>{amount}</h1><p>{symbol}</p>"
      />
    </styled.div>
    <styled.div
      title="Status"
    >
      UNKNOWN
    </styled.div>
  </styled.div>
  <styled.div>
    <span>
      2/3/2009
    </span>
  </styled.div>
</styled.div>
`;
